tests/test_diffusion.py

```diff
@@ -0,0 +1,199 @@
+                   time=time, transpose=True)
+
+    return {'U': ufield, 'V': vfield}
+
+
+def CreateStartField(lon, lat):
+    time = np.arange(0., 100000, 100000/2., dtype=np.float64)
+    # An evenly distributed starting density
+    data = np.ones((lon.size, lat.size, time.size), dtype=np.float32)
+    return data
+
+
+def LagrangianDiffusionNoCorrection(particle, grid, time, dt):
+    # Version of diffusion equation with no determenistic term i.e. brownian motion
+    to_lat = 1 / 1000. / 1.852 / 60.
+    to_lon = to_lat / math.cos(particle.lat*math.pi/180)
+    r_var = 1/3.
+    Rx = random.uniform(-1., 1.)
+    Ry = random.uniform(-1., 1.)
+    Kfield = grid.K[time, particle.lon, particle.lat]
+    Rx_component = Rx * math.sqrt(2 * Kfield * dt / r_var) * to_lon
+    Ry_component = Ry * math.sqrt(2 * Kfield * dt / r_var) * to_lat
+    particle.lon += Rx_component
+    particle.lat += Ry_component
+
+
+def UpdatePosition(particle, grid, time, dt):
+    particle.prev_lon = particle.new_lon
+    particle.new_lon = particle.lon
+    particle.prev_lat = particle.new_lat
+    particle.new_lat = particle.lat
+
+
+# Recovery Kernal for particles that diffuse outside boundary
+def Send2PreviousPoint(particle):
+    # print("Recovery triggered at %s | %s!" % (particle.lon, particle.lat))
+    # print("Moving particle back to %s | %s" % (particle.prev_lat, particle.prev_lat))
+    particle.lon = particle.prev_lon
+    particle.lat = particle.prev_lat
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def diffusion_test(mode, type='true_diffusion', particles=1000, timesteps=1000, output_file='diffusion_test'):
+    # Generate grid files according to given dimensions
+    gridx = 100
+    gridy = 100
+    # Generating grid with zero horizontal velocities
+    forcing_fields = CreateDummyUV(gridx, gridy)
+    grid = Grid(forcing_fields['U'], forcing_fields['V'], forcing_fields['U'].depth,
+                forcing_fields['U'].time, fields=forcing_fields)
+    # Create a non-uniform field of diffusivity
+    grid.add_field(CreateDiffusionField(grid))
+    # Calculate first differential of diffusivity field (needed for diffusion)
+    divK = grid.K.gradient()
+    grid.add_field(divK[0])
+    grid.add_field(divK[1])
+    # Calculate second differential (needed to estimate the appropriate minimum timestep to approximate Eulerian diffusion)
+    grid.add_field(grid.dK_dx.gradient(name="d2K")[0])
+    grid.add_field(grid.dK_dy.gradient(name="d2K")[1])
+
+    # Evenly spread starting distribution
+    Start_Field = Field('Start', CreateStartField(grid.U.lon, grid.U.lat),
+                        grid.U.lon, grid.U.lat, depth=grid.U.depth, time=grid.U.time, transpose=True)
+
+    # Calculate appropriate timestep
+    min_timestep = np.floor(np.min([np.min(1/np.abs(grid.d2K_dx.data)), np.min(1/np.abs(grid.d2K_dy.data))]))
+    print('Timestep should be < %s' % min_timestep)
+    timestep = 500
+
+    steps = timesteps
```

Comment:
This is actually a method to find the minimum timestep required to accurately replicate Eulerian diffusion- but you're right, it serves no purpose in this example, so I will remove it.